google/cloud/spanner_v1/pool.py

```diff
@@ -138,6 +163,113 @@ def session(self, **kwargs):
+        :param longRunningTransactionInterval_sec: Interval for running background task in seconds.
+        :param longRunningTransactionTimeout_sec: Timeout for recycling sessions in seconds.
+        """
+        long_running_transaction_timer = time.time()
+        transactions_closed = 0
+        while AbstractSessionPool._cleanup_task_ongoing_event.is_set():
+            is_timeout_reached = (
+                time.time() - long_running_transaction_timer
+                >= longRunningTransactionTimeout_sec
+                + longRunningTransactionInterval_sec
+            )
+
+            if is_timeout_reached and transactions_closed == 0:
+                break
+
+            AbstractSessionPool._cleanup_task_ongoing = True
+            iteration_start = time.time()
+
+            # Retrieve a list of sessions to delete
+            sessions_to_delete = [
+                session
+                for session in self._borrowed_sessions
+                if (datetime.datetime.utcnow() - session.checkout_time)
+                > datetime.timedelta(seconds=longRunningTransactionTimeout_sec)
+                and not session.long_running
+            ]
+
+            for session in sessions_to_delete:
+                self._recycle_session(session, transactions_closed)
+
+            # Calculate and sleep for the time remaining until the next iteration based on the interval.
+            iteration_elapsed = time.time() - iteration_start
+            remaining_time = longRunningTransactionInterval_sec - iteration_elapsed
+            if remaining_time > 0:
+                time.sleep(remaining_time)
+
+        AbstractSessionPool._cleanup_task_ongoing = False
+        AbstractSessionPool._cleanup_task_ongoing_event.clear()
+
+    def _recycle_session(self, session, transactions_closed):
+        """Helper method to recycle session for background task."""
+        session_trace = self._format_trace(self._traces[session._session_id])
+        if self._database.close_inactive_transactions:
+            if self._database.logging_enabled:
+                # Log a warning for a long-running transaction that has been closed
+                self._database.logger.warn(
+                    "Long running transaction! Transaction has been closed as it was running for "
+                    + "more than 60 minutes. For long running transactions use batch or partitioned transactions. "
+                    + session_trace
+                )
+
+            # Set the session as None for associated objects (transaction, batch, snapshot)
+            if session._transaction is not None:
+                session._transaction._session = None
+            if session._batch is not None:
+                session._batch._session = None
+            if session._snapshot is not None:
+                session._snapshot._session = None
+
+            # Increment the count of closed transactions and return the session to the pool
+            transactions_closed += 1
```

Comment:
integer values are passed by value and not reference so transaction_closed will remain 0 in deleteLongRunningTransactions. Objects are by reference. you can create an object for this